File: php/CyberAPI.php
<?php

namespace CyberAPI;

class CyberAPI {
    private $apiKey;
    private $baseUrl;
    private $timeout;

    public function __construct($apiKey, $baseUrl = 'https://threats.cyberapi.io/api/v1', $timeout = 10) {
        $this->apiKey = $apiKey;
        $this->baseUrl = rtrim($baseUrl, '/');
        $this->timeout = $timeout;
    }

    /**
     * Check a domain or IP for threats
     *
     * @param string|null $domain Domain name to check
     * @param string|null $ip IP address to check
     * @return array Response from the API
     * @throws \Exception If the request fails
     */
    public function check($domain = null, $ip = null) {
        if (!$domain && !$ip) {
            throw new \InvalidArgumentException("Either domain or ip must be provided");
        }

        $params = [];
        if ($domain) $params['domain'] = $domain;
        if ($ip) $params['ip'] = $ip;

        $url = $this->baseUrl . '/check?' . http_build_query($params);

        $ch = curl_init();
        curl_setopt($ch, CURLOPT_URL, $url);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_HTTPHEADER, [
            'X-API-Key: ' . $this->apiKey,
            'Accept: application/json',
            'User-Agent: CyberAPI-PHP/1.0.0'
        ]);
        curl_setopt($ch, CURLOPT_TIMEOUT, $this->timeout);

        $response = curl_exec($ch);
        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
        $error = curl_error($ch);
        
        curl_close($ch);

        if ($error) {
            throw new \Exception("Connection error: " . $error);
        }

        $data = json_decode($response, true);

        if ($httpCode >= 400) {
            $message = isset($data['detail']) ? $data['detail'] : "HTTP Error $httpCode";
            throw new \Exception($message, $httpCode);
        }

        return $data;
    }
}

File: go/cyberapi.go
package cyberapi

import (
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"net/url"
	"strings"
	"time"
)

const DefaultBaseURL = "https://threats.cyberapi.io/api/v1"

type Client struct {
	APIKey     string
	BaseURL    string
	HTTPClient *http.Client
}

func NewClient(apiKey string) *Client {
	return &Client{
		APIKey:  apiKey,
		BaseURL: DefaultBaseURL,
		HTTPClient: &http.Client{
			Timeout: 10 * time.Second,
		},
	}
}

type ThreatResponse struct {
	RiskScore int                    `json:"risk_score"`
	Verdict   string                 `json:"verdict"`
	Geo       map[string]interface{} `json:"geo,omitempty"`
	Signals   []string               `json:"signals,omitempty"`
	Details   map[string]interface{} `json:"details,omitempty"`
}

func (c *Client) Check(domain, ip string) (*ThreatResponse, error) {
	if domain == "" && ip == "" {
		return nil, fmt.Errorf("either domain or ip must be provided")
	}

	params := url.Values{}
	if domain != "" {
		params.Add("domain", domain)
	}
	if ip != "" {
		params.Add("ip", ip)
	}

	reqURL := fmt.Sprintf("%s/check?%s", strings.TrimRight(c.BaseURL, "/"), params.Encode())

	req, err := http.NewRequest("GET", reqURL, nil)
	if err != nil {
		return nil, err
	}

	req.Header.Set("X-API-Key", c.APIKey)
	req.Header.Set("Accept", "application/json")
	req.Header.Set("User-Agent", "CyberAPI-Go/1.0.0")

	resp, err := c.HTTPClient.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}

	if resp.StatusCode >= 400 {
		var errResp map[string]interface{}
		if err := json.Unmarshal(body, &errResp); err == nil {
			if detail, ok := errResp["detail"].(string); ok {
				return nil, fmt.Errorf("api error: %s", detail)
			}
		}
		return nil, fmt.Errorf("http error: %d", resp.StatusCode)
	}

	var result ThreatResponse
	if err := json.Unmarshal(body, &result); err != nil {
		return nil, err
	}

	return &result, nil
}
